lib/Serialization/Deserialization.cpp

```diff
@@ -1482,9 +1491,18 @@ ModuleFile::resolveCrossReference(ModuleDecl *baseModule, uint32_t pathLen) {
 
       pathTrace.addValue(memberName);
 
-      Type filterTy = getType(TID);
-      if (!isType)
+      Type filterTy;
```

Comment:
Is there a way to avoid the copy and paste here?

Reply:
The top-level part and the non-top-level part are pretty different, but I can see if there's enough in common to factor it out more than it is. It's still going to be ugly because of the need for the early return.

Reply:
Yeah, I don't think there's really a good way to avoid this. *Only* this section is copy/pasted, and because of the way `llvm::Error` works, you'd still end up with most of the duplication and a less clear building up of `pathTrace`. Maybe if we had a monadic `try` in C++.